### PR TITLE
[tests-only][full-ci] Add files drop upload e2e tests

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Added e2e test for uploading files via drag and drop action

## Related Issue
Part of https://github.com/owncloud/ocis/issues/5013

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- x ] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 1]
```diff
@@ -324,6 +328,20 @@ export const uploadResource = async (args: uploadResourceArgs): Promise<void> =>
   }
 }
 
+export const dropUploadFiles = async (args: uploadResourceArgs): Promise<void> => {
```
Comment (amrita-shrestha) [inline tests/e2e/support/objects/app-files/resource/actions.ts:331]: does this function only drop files?

Comment (saw-jan) [reply tests/e2e/support/objects/app-files/resource/actions.ts:331]: yes. only files

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 2]
```diff
@@ -324,6 +328,20 @@ export const uploadResource = async (args: uploadResourceArgs): Promise<void> =>
...
+  const { page, resources } = args
+
+  await dragDropFiles(page, resources, filesView)
+
+  await page.waitForSelector(uploadInfoSuccessLabelSelector)
+  await page.locator(uploadInfoShowDetailsButton).click()
+  await Promise.all(
```
Comment (ScharfViktor) [inline tests/e2e/support/objects/app-files/resource/actions.ts:338]: does it check all uploaded files?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 3]
```diff
@@ -324,6 +328,20 @@ export const uploadResource = async (args: uploadResourceArgs): Promise<void> =>
...
+    resources.map((file) =>
+      page.waitForSelector(util.format(resourceNameSelector, path.basename(file.name)))
```
Comment (ScharfViktor) [inline tests/e2e/support/objects/app-files/resource/actions.ts:340]: if you go upload info maybe you should check particular item

Screenshot 2023-05-22 at 10 58 38

**Review by ScharfViktor — COMMENTED** [review]
LGTM

**Review by ScharfViktor — COMMENTED** [review]
LGTM

File: tests/e2e/support/objects/app-files/resource/actions.ts (hunk 2 above)
Comment (saw-jan) [reply tests/e2e/support/objects/app-files/resource/actions.ts:338]: Yeah, checks all the uploaded resources

File: tests/e2e/support/objects/app-files/resource/actions.ts (hunk 3 above)
Comment (saw-jan) [reply tests/e2e/support/objects/app-files/resource/actions.ts:340]: The selector is not strict. I will fix it

**Review by ScharfViktor — APPROVED** [review]

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/35847/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2023-5-22-10-22-25.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2023-5-22-10-22-56.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/file-alice-2023-5-22-10-34-42.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/file-brian-2023-5-22-10-35-05.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/folder-alice-2023-5-22-10-31-21.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/folder-brian-2023-5-22-10-31-33.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/share-a-link-with-internal-role-alice-2023-5-22-10-25-39.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/share-a-link-with-internal-role-brian-2023-5-22-10-25-49.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/tag-sharing-alice-2023-5-22-10-42-23.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/tag-sharing-brian-2023-5-22-10-42-33.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/unstructured-collection-of-testable-space-interactions-alice-2023-5-22-10-39-25.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/unstructured-collection-of-testable-space-interactions-anonymous-2023-5-22-10-40-14.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/unstructured-collection-of-testable-space-interactions-brian-2023-5-22-10-40-15.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/35847/tracing/users-can-navigate-web-via-tiles-alice-2023-5-22-10-43-38.zip 


</p></details>

**Review by grgprarup — COMMENTED** [review]
lgtm :+1:

File: tests/e2e/support/objects/app-files/page/public.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ import {
 
 const passwordInput = 'input[type="password"]'
 const fileUploadInput = '//input[@id="files-file-upload-input"]'
-const resourceNameSelector = '[data-test-resource-name="%s"]'
+const resourceNameSelector = '.upload-info-items [data-test-resource-name="%s"]'
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/e2e/support/objects/app-files/page/public.ts:19]: the selector is for the public link uploader role page. selector name might come from copy-paste. yeah, good to have it renamed. :+1:

**Review by SwikritiT — APPROVED** [review]
LGTM

**Pull request merged**